Import pandas

In [1]:
import pandas as pd
pd.set_option('display.max_rows', 200)

Carregar o dataframe `escolas` do arquivo csv.

In [2]:
escolas = pd.read_csv('data/escolas.csv')
escolas

id                                     Escolas_Postos  \
0    178        CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5     17                       E.M PRESIDENTE JOSE LINHARES   
6     55                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9     89                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CÍCERO PENA   
14   413                                  EM CÓCIO BARCELOS   
15   413                                  EM CÓCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MÉXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
20    17                       E.M PRESIDENTE JOSE LINHARES   
21    35                                  EM SENADOR CORREA   
22    11                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRÉ   
25    89                                   EM MARIO CLAUDIO   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31    60               ESCOLA MUNICIPAL BARÃO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARÉ   
33   653                            EM LÚCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPÉIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39    25                     ESCOLA MUNICIPAL GEORGE SUMMER   
40   660                       E.M NOSSA SENHORA DA POMPÉIA   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44    20                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA VINÍCIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53    41                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
57    89                                   EM MARIO CLAUDIO   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
60   388                                  EM CARDEAL CAMARA   
61   498                                           EM CHILE   
62   663                          EM 

Renomear as colunas de `escolas` e aplicar o padrão snake case.

In [3]:
column_mapping = {
    'id': 'id',
    'Escolas_Postos': 'nome_escola',
    'BAIRRO': 'bairro',
    'ENDEREÇO ': 'logradouro_entrega',
    'lat': 'latitude',
    'lon': 'longitude'
}

escolas.rename(columns=column_mapping,inplace=True)
escolas

id                                        nome_escola  \
0    178        CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5     17                       E.M PRESIDENTE JOSE LINHARES   
6     55                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9     89                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CÍCERO PENA   
14   413                                  EM CÓCIO BARCELOS   
15   413                                  EM CÓCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MÉXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
20    17                       E.M PRESIDENTE JOSE LINHARES   
21    35                                  EM SENADOR CORREA   
22    11                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRÉ   
25    89                                   EM MARIO CLAUDIO   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31    60               ESCOLA MUNICIPAL BARÃO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARÉ   
33   653                            EM LÚCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPÉIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39    25                     ESCOLA MUNICIPAL GEORGE SUMMER   
40   660                       E.M NOSSA SENHORA DA POMPÉIA   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44    20                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA VINÍCIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53    41                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
57    89                                   EM MARIO CLAUDIO   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
60   388                                  EM CARDEAL CAMARA   
61   498                                           EM CHILE   
62   663                          EM 

Mudar o tipo de dados da coluna `id`para string e preencher com zeros (à esquerda) até len = 3.

In [4]:
escolas['id'] = escolas['id'].astype(str).str.zfill(3)
escolas

id                                        nome_escola  \
0    178        CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5    017                       E.M PRESIDENTE JOSE LINHARES   
6    055                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9    089                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CÍCERO PENA   
14   413                                  EM CÓCIO BARCELOS   
15   413                                  EM CÓCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MÉXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
20   017                       E.M PRESIDENTE JOSE LINHARES   
21   035                                  EM SENADOR CORREA   
22   011                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRÉ   
25   089                                   EM MARIO CLAUDIO   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31   060               ESCOLA MUNICIPAL BARÃO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARÉ   
33   653                            EM LÚCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPÉIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39   025                     ESCOLA MUNICIPAL GEORGE SUMMER   
40   660                       E.M NOSSA SENHORA DA POMPÉIA   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44   020                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA VINÍCIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53   041                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
57   089                                   EM MARIO CLAUDIO   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
60   388                                  EM CARDEAL CAMARA   
61   498                                           EM CHILE   
62   663                          EM 

Substituir os valores da coluna `bairro` em `escolas` com os valores corretos do dataframe `subprefeituras`.
Para alguns valores, como 'Maracanã/ Vila Isabel', 'Maracanã/ Tijuca', 'Recreio', 'Rio Das Pedras', 'Freguesia Jpa', 'Augusto Vasconcelos' e 'Nova Sepetiba', foi feita busca na internet para identificar o bairro correto.

In [5]:
bairros = {
'HUMAITA':'HUMAITÁ', 
'MARACANÃ/ VILA ISABEL':'VILA ISABEL',
'MARACANÃ/ TIJUCA':'VILA ISABEL',
'MEIER':'MÉIER', 
'INHAUMA':'INHAÚMA',
'OSWALDO CRUZ':'OSVALDO CRUZ',
'GARDENIA AZUL':'GARDÊNIA AZUL',
'RECREIO':'RECREIO DOS BANDEIRANTES',
'RIO DAS PEDRAS':'JACAREPAGUÁ',
'FREGUESIA JPA': 'FREGUESIA (JACAREPAGUÁ)',
'AUGUSTO VASCONCELOS':'SENADOR VASCONCELOS',
'PACIENCIA':'PACIÊNCIA',
'NOVA SEPETIBA': 'SEPETIBA'
}

escolas1 = escolas
escolas1['bairro'] = escolas1['bairro'].replace(bairros)
escolas1

id                                        nome_escola  \
0    178        CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5    017                       E.M PRESIDENTE JOSE LINHARES   
6    055                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9    089                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CÍCERO PENA   
14   413                                  EM CÓCIO BARCELOS   
15   413                                  EM CÓCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MÉXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
20   017                       E.M PRESIDENTE JOSE LINHARES   
21   035                                  EM SENADOR CORREA   
22   011                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRÉ   
25   089                                   EM MARIO CLAUDIO   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31   060               ESCOLA MUNICIPAL BARÃO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARÉ   
33   653                            EM LÚCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPÉIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39   025                     ESCOLA MUNICIPAL GEORGE SUMMER   
40   660                       E.M NOSSA SENHORA DA POMPÉIA   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44   020                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA VINÍCIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53   041                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
57   089                                   EM MARIO CLAUDIO   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
60   388                                  EM CARDEAL CAMARA   
61   498                                           EM CHILE   
62   663                          EM 

Transformar todas as colunas com tipo de dados string em `escolas` para upper case.

In [6]:
escolas2 = escolas1.apply(lambda x:x.astype(str).str.upper())
escolas2

id                                        nome_escola  \
0    178        CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5    017                       E.M PRESIDENTE JOSE LINHARES   
6    055                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9    089                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CÍCERO PENA   
14   413                                  EM CÓCIO BARCELOS   
15   413                                  EM CÓCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MÉXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
20   017                       E.M PRESIDENTE JOSE LINHARES   
21   035                                  EM SENADOR CORREA   
22   011                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRÉ   
25   089                                   EM MARIO CLAUDIO   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31   060               ESCOLA MUNICIPAL BARÃO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARÉ   
33   653                            EM LÚCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPÉIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39   025                     ESCOLA MUNICIPAL GEORGE SUMMER   
40   660                       E.M NOSSA SENHORA DA POMPÉIA   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44   020                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA VINÍCIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53   041                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
57   089                                   EM MARIO CLAUDIO   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
60   388                                  EM CARDEAL CAMARA   
61   498                                           EM CHILE   
62   663                          EM 

Remover acentos de todas as colunas com tipo de dados string em `escolas`.

In [7]:
cols = escolas2.select_dtypes(include=[object]).columns
escolas3 = escolas2
escolas3[cols] = escolas2.apply(lambda x:x.str.normalize('NFKD').str.encode('ascii',errors='ignore').str.decode('utf-8'))
escolas3

id                                        nome_escola  \
0    178        CENTRO INTEGRADO DE EDUCACAO PUBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5    017                       E.M PRESIDENTE JOSE LINHARES   
6    055                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9    089                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CICERO PENA   
14   413                                  EM COCIO BARCELOS   
15   413                                  EM COCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MEXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
20   017                       E.M PRESIDENTE JOSE LINHARES   
21   035                                  EM SENADOR CORREA   
22   011                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRE   
25   089                                   EM MARIO CLAUDIO   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31   060               ESCOLA MUNICIPAL BARAO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARE   
33   653                            EM LUCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPEIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39   025                     ESCOLA MUNICIPAL GEORGE SUMMER   
40   660                       E.M NOSSA SENHORA DA POMPEIA   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44   020                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCACAO PUBLICA VINICIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53   041                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
57   089                                   EM MARIO CLAUDIO   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
60   388                                  EM CARDEAL CAMARA   
61   498                                           EM CHILE   
62   663                          EM 

Identificar registros duplicados em `escolas`.

In [8]:
duplicates_mask = escolas3.duplicated(subset='id',keep=False)
duplicates_df = escolas3[duplicates_mask]
duplicates_df.sort_values('id')

id                         nome_escola           bairro  \
5    017        E.M PRESIDENTE JOSE LINHARES          IPANEMA   
20   017        E.M PRESIDENTE JOSE LINHARES          IPANEMA   
135  069  ESCOLA MUNICIPAL CARDEAL ARCOVERDE        MADUREIRA   
80   069  ESCOLA MUNICIPAL CARDEAL ARCOVERDE        MADUREIRA   
143  069  ESCOLA MUNICIPAL CARDEAL ARCOVERDE        MADUREIRA   
69   069  ESCOLA MUNICIPAL CARDEAL ARCOVERDE        MADUREIRA   
151  069  ESCOLA MUNICIPAL CARDEAL ARCOVERDE        MADUREIRA   
9    089                    EM MARIO CLAUDIO     RIO COMPRIDO   
57   089                    EM MARIO CLAUDIO     RIO COMPRIDO   
25   089                    EM MARIO CLAUDIO     RIO COMPRIDO   
84   089                    EM MARIO CLAUDIO     RIO COMPRIDO   
70   090                 EM THOMAS JEFFERSON            ACARI   
78   090                 EM THOMAS JEFFERSON            ACARI   
117  206                         EM HONDURAS       PRACA SECA   
99   206                         EM HONDURAS       PRACA SECA   
56   379                 CIEP NELSON MANDELA     CAMPO GRANDE   
123  379                 CIEP NELSON MANDELA     CAMPO GRANDE   
112  379                 CIEP NELSON MANDELA     CAMPO GRANDE   
49   388                   EM CARDEAL CAMARA  PARADA DE LUCAS   
60   388                   EM CARDEAL CAMARA  PARADA DE LUCAS   
14   413                   EM COCIO BARCELOS       COPACABANA   
15   413                   EM COCIO BARCELOS       COPACABANA   
85   520             EM BARONESA DE SAAVEDRA         REALENGO   
109  520             EM BARONESA DE SAAVEDRA         REALENGO   
40   660        E.M NOSSA SENHORA DA POMPEIA  TODOS OS SANTOS   
34   660        E.M NOSSA SENHORA DA POMPEIA  TODOS OS SANTOS   

                            logradouro_entrega    latitude   longitude  
5                        R. BARAO DA TORRE, 90  -22,983332  -43,199036  
20                       R. BARAO DA TORRE, 90  -22,983332  -43,199036  
135                 RUA AGOSTINHO BARBALHO 401  -22,876082  -43,343369  
80                  RUA AGOSTINHO BARBALHO 401  -22,876082  -43,343369  
143                 RUA AGOSTINHO BARBALHO 401  -22,876082  -43,343369  
69                  RUA AGOSTINHO BARBALHO 401  -22,876082  -43,343369  
151                 RUA AGOSTINHO BARBALHO 401  -22,876082  -43,343369  
9                         R. HADDOCK LOBO, 148  -22,917233  -43,211044  
57                        R. HADDOCK LOBO, 148  -22,917233  -43,211044  
25                        R. HADDOCK LOBO, 148  -22,917233  -43,211044  
84                        R. HADDOCK LOBO, 148  -22,917233  -43,211044  
70   AVENIDA PASTOR MARTIN LUTHER KING JR S/No   -22,81631  -43,340149  
78   AVENIDA PASTOR MARTIN LUTHER KING JR S/No   -22,81631  -43,340149  
117                  PRACA BARAO DA TAQUARA 12   -22896931   -43352667  
99                   PRACA BARAO DA TAQUARA 12  -22,896931  -43,352667  
56                     PRACA DANIEL LAMES, S/N  -22,895575  -43,570508  
123                   PRACA DANIEL LAMES, S/No  -22,895575  -43,570508  
112                   PRACA DANIEL LAMES, S/No  -22,895575  -43,570508  
49                                R. ANAMA, 98   -22,81527  -43,303566  
60                                R. ANAMA, 98   -22,81527  -43,303566  
14                     R. BARAO DE IPANEMA, 34  -22,974828  -43,189063  
15                     R. BARAO DE IPANEMA, 34   -22,74828  -43,189063  
85           R. MAL. BIBIANO COSTALAT, 483-639   -22,87185  -43,432432  
109          R. MAL. BIBIANO COSTALAT, 483-639   -22,87185  -43,432432  
40                          R. CIRNE MAIA, 109  -22,892031  -43,279925  
34                          R. CIRNE MAIA, 109  -22,892031  -43,279925

Eliminar registros duplicados de `escolas`.
Os registros de índices 117, 112 e 15 não foram eliminados por conter ligeiras diferenças com suas contrapartes, como a falta de uma vírgula, a presença de um caracter especial ou um erro de digitação. Dessa forma, sua eliminação foi manual.

* 117	206	EM HONDURAS - sem vírgula na latitude e longitude
* 112	379	CIEP NELSON MANDELA - º no endereço
* 15	413	EM CÓCIO BARCELOS - latitude incompleta

In [9]:
escolas4 = escolas3.drop_duplicates()
escolas4 = escolas4.drop([117,112,15]) # elimina escolas duplicadas, mas com ligeira diferença no registro
escolas4

id                                        nome_escola  \
0    178        CENTRO INTEGRADO DE EDUCACAO PUBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5    017                       E.M PRESIDENTE JOSE LINHARES   
6    055                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9    089                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CICERO PENA   
14   413                                  EM COCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MEXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
21   035                                  EM SENADOR CORREA   
22   011                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRE   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31   060               ESCOLA MUNICIPAL BARAO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARE   
33   653                            EM LUCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPEIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39   025                     ESCOLA MUNICIPAL GEORGE SUMMER   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44   020                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCACAO PUBLICA VINICIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53   041                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
61   498                                           EM CHILE   
62   663                          EM CIENTISTA MARIO KROEFF   
63   121                                 EM DILERMANDO CRUZ   
64   161                                   E.M JOAO DE DEUS   
65   263                         EM PADRE MANUEL DE NOBREGA   
66   118   CENTRO INTEGRADO DE EDUCACAO PUBLICA ELIS REGINA   
67   140                                     EM PEDRO LESSA   
68   260                       EM PRO

Arredondar os campos 'latitude' e 'longitude' para cinco casas decimais.

In [10]:
escolas5 = escolas4
escolas5['latitude'] = escolas5['latitude'].str.replace(',', '.').astype(float).round(5)
escolas5['longitude'] = escolas5['longitude'].str.replace(',', '.').astype(float).round(5)

escolas5

id                                        nome_escola  \
0    178        CENTRO INTEGRADO DE EDUCACAO PUBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5    017                       E.M PRESIDENTE JOSE LINHARES   
6    055                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9    089                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CICERO PENA   
14   413                                  EM COCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MEXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
21   035                                  EM SENADOR CORREA   
22   011                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRE   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31   060               ESCOLA MUNICIPAL BARAO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARE   
33   653                            EM LUCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPEIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39   025                     ESCOLA MUNICIPAL GEORGE SUMMER   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44   020                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCACAO PUBLICA VINICIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53   041                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
61   498                                           EM CHILE   
62   663                          EM CIENTISTA MARIO KROEFF   
63   121                                 EM DILERMANDO CRUZ   
64   161                                   E.M JOAO DE DEUS   
65   263                         EM PADRE MANUEL DE NOBREGA   
66   118   CENTRO INTEGRADO DE EDUCACAO PUBLICA ELIS REGINA   
67   140                                     EM PEDRO LESSA   
68   260                       EM PRO

Criar coluna `tipo_escola`

In [11]:
tipo_escola = {
'CENTRO INTEGRADO DE EDUCACAO PUBLICA':'CIEP', 
'E.M ':'EM ',
'E.M. ':'EM ',
'EM. ':'EM ', 
'ESCOLA MUNICIPAL ':'EM '
}

escolas6 = escolas5
for key, value in tipo_escola.items():
    escolas6['nome_escola'] = escolas6['nome_escola'].str.replace(fr'(?i)\b{key}\b', value, regex=True)

escolas6

id                                  nome_escola  \
0    178                                  CIEP HENFIL   
1    634                   EM ALICE DO AMARAL PEIXOTO   
2    483                           EM CELESTINO SILVA   
3    476                          EM FLORIANO PEIXOTO   
4    132                            EM PEREIRA PASSOS   
5    017                  EM PRESIDENTE JOSE LINHARES   
6    055                            EM SANTA CATARINA   
7    490                                 EM TIA CIATA   
8    600                                   EM URUGUAI   
9    089                             EM MARIO CLAUDIO   
10   387               CIEP PRESIDENTE AGOSTINHO NETO   
11   331                          CIEP TANCREDO NEVES   
12   450                             EM ALBERTO BARTH   
13   558                               EM CICERO PENA   
14   413                            EM COCIO BARCELOS   
16   540                        EM HENRIQUE DODSWORTH   
17   309                                    EM MEXICO   
18   268                              EM MINAS GERAIS   
19   523               EM PRESIDENTE DA COSTA E SILVA   
21   035                            EM SENADOR CORREA   
22   011                           CIEP SAMUEL WEINER   
23   226                           EM AFRANIO PEIXOTO   
24   638                             EM AZEVEDO SODRE   
26   126                           EM DUQUE DE CAXIAS   
27   180                                   EM EQUADOR   
28   163                           EM LAUDIMIA TROTTA   
29   361                        EM PRUDENTE DE MORAES   
30   280           EM GENERAL HUMBERTO DE SOUZA MELLO   
31   060                       EM BARAO HOMEM DE MELO   
32   587                       EM ALMIRANTE TAMANDARE   
33   653                      EM LUCIA MIGUEL PEREIRA   
34   660                  EM NOSSA SENHORA DA POMPEIA   
35   382                              EM SANTOS ANJOS   
36   217                                   EM ALAGOAS   
37   378                     EM BRIGADEIRO FARIA LIMA   
38   429                             EM FELIX PACHECO   
39   025                             EM GEORGE SUMMER   
41   373                 EM PROFESSOR AUGUSTO PAULINO   
42   624                         EM RIO GRANDE DO SUL   
43   157                                 EM SARMIENTO   
44   020                                    EM TAGORE   
45   117                  CIEP CHANCELER WILLY BRANDT   
46   502                         CIEP PATRICE LUMUMBA   
47   351                      CIEP VINICIUS DE MORAES   
48   135                          EM DOMINGOS BEBIANO   
49   388                            EM CARDEAL CAMARA   
50   440                           EM JOAQUIM RIBEIRO   
51   259                                EM PERNAMBUCO   
52   422                         EM ALBINO SOUZA CRUZ   
53   041                             EM ANDRADE NEVES   
54   537                             EM ARY QUINTELLA   
55   230                                    EM BERLIM   
56   379                          CIEP NELSON MANDELA   
58   146                      EM BERNARDO VASCONCELOS   
59   566                               EM BRANT HORTA   
61   498                                     EM CHILE   
62   663                    EM CIENTISTA MARIO KROEFF   
63   121                           EM DILERMANDO CRUZ   
64   161                              EM JOAO DE DEUS   
65   263                   EM PADRE MANUEL DE NOBREGA   
66   118                             CIEP ELIS REGINA   
67   140                               EM PEDRO LESSA   
68   260                 EM PROFESSOR JOSUE DE CASTRO   
69   069                         EM CARDEAL ARCOVERDE   
70   090                          EM THOMAS JEFFERSON   
71   274                            EM ABEILARD FEIJO   
72   676                           EM ANISIO TEIXEIRA   
73   390                     EM ANDREA FONTES PEIXOTO   
74   608                     EM GRANDJEAN DE MONTIGNY   
75   589        

Criar coluna `tipo_escola` com valores extraídos de `escolas_postos` (EM|CIEP|COLEGIO MUNICIPAL).

In [12]:
escolas7 = escolas6
escolas7['tipo_escola'] = escolas7['nome_escola'].str.extract(r'(EM|CIEP|COLEGIO MUNICIPAL)', expand=False)
escolas7['nome_escola'] = escolas7['nome_escola'].str.replace(r'(EM|CIEP|COLEGIO MUNICIPAL)', '', regex=True).str.strip()
cols = ['id','nome_escola','tipo_escola','logradouro_entrega','bairro','latitude','longitude']
escolas7 = escolas7[cols]
escolas7

id                              nome_escola        tipo_escola  \
0    178                                   HENFIL               CIEP   
1    634                  ALICE DO AMARAL PEIXOTO                 EM   
2    483                          CELESTINO SILVA                 EM   
3    476                         FLORIANO PEIXOTO                 EM   
4    132                           PEREIRA PASSOS                 EM   
5    017                 PRESIDENTE JOSE LINHARES                 EM   
6    055                           SANTA CATARINA                 EM   
7    490                                TIA CIATA                 EM   
8    600                                  URUGUAI                 EM   
9    089                            MARIO CLAUDIO                 EM   
10   387                PRESIDENTE AGOSTINHO NETO               CIEP   
11   331                           TANCREDO NEVES               CIEP   
12   450                            ALBERTO BARTH                 EM   
13   558                              CICERO PENA                 EM   
14   413                           COCIO BARCELOS                 EM   
16   540                       HENRIQUE DODSWORTH                 EM   
17   309                                   MEXICO                 EM   
18   268                             MINAS GERAIS                 EM   
19   523              PRESIDENTE DA COSTA E SILVA                 EM   
21   035                           SENADOR CORREA                 EM   
22   011                            SAMUEL WEINER               CIEP   
23   226                          AFRANIO PEIXOTO                 EM   
24   638                            AZEVEDO SODRE                 EM   
26   126                          DUQUE DE CAXIAS                 EM   
27   180                                  EQUADOR                 EM   
28   163                          LAUDIMIA TROTTA                 EM   
29   361                       PRUDENTE DE MORAES                 EM   
30   280          GENERAL HUMBERTO DE SOUZA MELLO                 EM   
31   060                        BARAO HOM DE MELO                 EM   
32   587                      ALMIRANTE TAMANDARE                 EM   
33   653                     LUCIA MIGUEL PEREIRA                 EM   
34   660                 NOSSA SENHORA DA POMPEIA                 EM   
35   382                             SANTOS ANJOS                 EM   
36   217                                  ALAGOAS                 EM   
37   378                    BRIGADEIRO FARIA LIMA                 EM   
38   429                            FELIX PACHECO                 EM   
39   025                            GEORGE SUMMER                 EM   
41   373                PROFESSOR AUGUSTO PAULINO                 EM   
42   624                        RIO GRANDE DO SUL                 EM   
43   157                                SARMIENTO                 EM   
44   020                                   TAGORE                 EM   
45   117                   CHANCELER WILLY BRANDT               CIEP   
46   502                          PATRICE LUMUMBA               CIEP   
47   351                       VINICIUS DE MORAES               CIEP   
48   135                         DOMINGOS BEBIANO                 EM   
49   388                           CARDEAL CAMARA                 EM   
50   440                          JOAQUIM RIBEIRO                 EM   
51   259                               PERNAMBUCO                 EM   
52   422                        ALBINO SOUZA CRUZ                 EM   
53   041                            ANDRADE NEVES                 EM   
54   537                            ARY QUINTELLA                 EM   
55   230                                   BERLIM                 EM   
56   379                           NELSON MANDELA               CIEP   
58   146                     BERNARDO VASCONCELOS                 EM   
59   566                              BRANT HO

Corrige as abreviações nos endereços.

In [13]:
correcoes_logradouro = {
    'AV ':'AVENIDA ', 
    'AV. ':'AVENIDA ',
    'CONJ. ':'CONJUNTO ',
    'CONJ ':'CONJUNTO ',
    'ESTR. ':'ESTRADA ',
    'PCA. ': 'PRACA ',
    'R. ':'RUA ',
    'ALM. ':'ALMIRANTE ',
    'ENG. ':'ENGENHEIRO ',
    'MAL. ':'MARECHAL ',
    'SD. ':'SOLDADO ',
    'SEN. ':'SENADOR ',
    'S/No':'S/N', 
    'S/N.o':'S/N',
}

escolas8 = escolas7
for key, value in correcoes_logradouro.items():
    escolas8['logradouro_entrega'] = escolas8['logradouro_entrega'].str.replace(fr'(?i)\b{key}\b', value, regex=True)

escolas8

id                              nome_escola        tipo_escola  \
0    178                                   HENFIL               CIEP   
1    634                  ALICE DO AMARAL PEIXOTO                 EM   
2    483                          CELESTINO SILVA                 EM   
3    476                         FLORIANO PEIXOTO                 EM   
4    132                           PEREIRA PASSOS                 EM   
5    017                 PRESIDENTE JOSE LINHARES                 EM   
6    055                           SANTA CATARINA                 EM   
7    490                                TIA CIATA                 EM   
8    600                                  URUGUAI                 EM   
9    089                            MARIO CLAUDIO                 EM   
10   387                PRESIDENTE AGOSTINHO NETO               CIEP   
11   331                           TANCREDO NEVES               CIEP   
12   450                            ALBERTO BARTH                 EM   
13   558                              CICERO PENA                 EM   
14   413                           COCIO BARCELOS                 EM   
16   540                       HENRIQUE DODSWORTH                 EM   
17   309                                   MEXICO                 EM   
18   268                             MINAS GERAIS                 EM   
19   523              PRESIDENTE DA COSTA E SILVA                 EM   
21   035                           SENADOR CORREA                 EM   
22   011                            SAMUEL WEINER               CIEP   
23   226                          AFRANIO PEIXOTO                 EM   
24   638                            AZEVEDO SODRE                 EM   
26   126                          DUQUE DE CAXIAS                 EM   
27   180                                  EQUADOR                 EM   
28   163                          LAUDIMIA TROTTA                 EM   
29   361                       PRUDENTE DE MORAES                 EM   
30   280          GENERAL HUMBERTO DE SOUZA MELLO                 EM   
31   060                        BARAO HOM DE MELO                 EM   
32   587                      ALMIRANTE TAMANDARE                 EM   
33   653                     LUCIA MIGUEL PEREIRA                 EM   
34   660                 NOSSA SENHORA DA POMPEIA                 EM   
35   382                             SANTOS ANJOS                 EM   
36   217                                  ALAGOAS                 EM   
37   378                    BRIGADEIRO FARIA LIMA                 EM   
38   429                            FELIX PACHECO                 EM   
39   025                            GEORGE SUMMER                 EM   
41   373                PROFESSOR AUGUSTO PAULINO                 EM   
42   624                        RIO GRANDE DO SUL                 EM   
43   157                                SARMIENTO                 EM   
44   020                                   TAGORE                 EM   
45   117                   CHANCELER WILLY BRANDT               CIEP   
46   502                          PATRICE LUMUMBA               CIEP   
47   351                       VINICIUS DE MORAES               CIEP   
48   135                         DOMINGOS BEBIANO                 EM   
49   388                           CARDEAL CAMARA                 EM   
50   440                          JOAQUIM RIBEIRO                 EM   
51   259                               PERNAMBUCO                 EM   
52   422                        ALBINO SOUZA CRUZ                 EM   
53   041                            ANDRADE NEVES                 EM   
54   537                            ARY QUINTELLA                 EM   
55   230                                   BERLIM                 EM   
56   379                           NELSON MANDELA               CIEP   
58   146                     BERNARDO VASCONCELOS                 EM   
59   566                              BRANT HO

Separa número do imível do logradouro, remove pontos e números do logradouro e reodena colunas.

In [14]:
escolas9 = escolas8

escolas9['numero'] = escolas9['logradouro_entrega'].str.extract(r'(\d+)')
escolas9['logradouro_entrega'] = escolas9['logradouro_entrega'].str.replace(r'(\d+)', '',regex=True).str.strip()
escolas9['logradouro_entrega'] = escolas9['logradouro_entrega'].str.replace(r'[,.]', '',regex=True).str.strip()
escolas9['logradouro_entrega'] = escolas9['logradouro_entrega'].str.replace(r'( S/N|S/N )', '',regex=True).str.strip()

cols = ['id','nome_escola','tipo_escola','logradouro_entrega','numero','bairro','latitude','longitude']
escolas9 = escolas9[cols]

escolas9

id                              nome_escola        tipo_escola  \
0    178                                   HENFIL               CIEP   
1    634                  ALICE DO AMARAL PEIXOTO                 EM   
2    483                          CELESTINO SILVA                 EM   
3    476                         FLORIANO PEIXOTO                 EM   
4    132                           PEREIRA PASSOS                 EM   
5    017                 PRESIDENTE JOSE LINHARES                 EM   
6    055                           SANTA CATARINA                 EM   
7    490                                TIA CIATA                 EM   
8    600                                  URUGUAI                 EM   
9    089                            MARIO CLAUDIO                 EM   
10   387                PRESIDENTE AGOSTINHO NETO               CIEP   
11   331                           TANCREDO NEVES               CIEP   
12   450                            ALBERTO BARTH                 EM   
13   558                              CICERO PENA                 EM   
14   413                           COCIO BARCELOS                 EM   
16   540                       HENRIQUE DODSWORTH                 EM   
17   309                                   MEXICO                 EM   
18   268                             MINAS GERAIS                 EM   
19   523              PRESIDENTE DA COSTA E SILVA                 EM   
21   035                           SENADOR CORREA                 EM   
22   011                            SAMUEL WEINER               CIEP   
23   226                          AFRANIO PEIXOTO                 EM   
24   638                            AZEVEDO SODRE                 EM   
26   126                          DUQUE DE CAXIAS                 EM   
27   180                                  EQUADOR                 EM   
28   163                          LAUDIMIA TROTTA                 EM   
29   361                       PRUDENTE DE MORAES                 EM   
30   280          GENERAL HUMBERTO DE SOUZA MELLO                 EM   
31   060                        BARAO HOM DE MELO                 EM   
32   587                      ALMIRANTE TAMANDARE                 EM   
33   653                     LUCIA MIGUEL PEREIRA                 EM   
34   660                 NOSSA SENHORA DA POMPEIA                 EM   
35   382                             SANTOS ANJOS                 EM   
36   217                                  ALAGOAS                 EM   
37   378                    BRIGADEIRO FARIA LIMA                 EM   
38   429                            FELIX PACHECO                 EM   
39   025                            GEORGE SUMMER                 EM   
41   373                PROFESSOR AUGUSTO PAULINO                 EM   
42   624                        RIO GRANDE DO SUL                 EM   
43   157                                SARMIENTO                 EM   
44   020                                   TAGORE                 EM   
45   117                   CHANCELER WILLY BRANDT               CIEP   
46   502                          PATRICE LUMUMBA               CIEP   
47   351                       VINICIUS DE MORAES               CIEP   
48   135                         DOMINGOS BEBIANO                 EM   
49   388                           CARDEAL CAMARA                 EM   
50   440                          JOAQUIM RIBEIRO                 EM   
51   259                               PERNAMBUCO                 EM   
52   422                        ALBINO SOUZA CRUZ                 EM   
53   041                            ANDRADE NEVES                 EM   
54   537                            ARY QUINTELLA                 EM   
55   230                                   BERLIM                 EM   
56   379                           NELSON MANDELA               CIEP   
58   146                     BERNARDO VASCONCELOS                 EM   
59   566                              BRANT HO

Carregar o dataframe `material_didatico` do arquivo csv.

In [15]:
material_didatico = pd.read_csv('data/material_didatico.csv')
material_didatico

id Quantidade
0    178         20
1    634        121
2    483        220
3    476        190
4    132         67
5    410        397
6     55         92
7    490        578
8    600        591
9     89        473
10   387        456
11   331        223
12   450        420
13   558        573
14   413         ''
15   698        366
16   540        516
17   309        319
18   268        308
19   523        111
20    17         48
21    35        525
22    11        445
23   226        164
24   638        293
25   271        152
26   126        399
27   180        185
28   163        576
29   361         63
30   280        491
31    60      'NaN'
32   587        336
33   653        159
34   660        342
35   382        262
36   217        440
37   378        365
38   429        NaN
39    25        183
40   367        297
41   373         44
42   624        109
43   157        350
44    20        255
45   117        457
46   502        493
47   351        489
48   135        456
49   647         64
50   440        530
51   259        158
52   422        123
53    41        468
54   537        234
55   230         24
56   683        218
57   146        151
58   566        564
59   388        504
60   498         61
61   663        426
62   121        167
63   161        509
64   263        292
65   118        210
66   140        153
67   260      '397'
68   236         41
69    90         67
70   274        561
71   676        180
72   390        257
73   608        260
74   589        317
75   256        345
76    91      'NaN'
77   285         32
78   248        513
79   441        338
80   623        416
81   603        352
82   673        393
83   182        425
84   520        136
85   412      '395'
86    31        115
87   514        534
88    23        342
89   415        152
90    16        430
91   570        466
92   528         51
93   414         26
94   690         94
95   202        121
96   550        128
97   211        520
98   526        524
99   492        NaN
100  466        279
101  636        161
102  527        179
103  586        194
104   80        599
105  398        315
106  175        435
107  308        400
108  512        294
109  576        148
110  495        128
111  379        109
112  641        269
113  689        571
114  171        235
115  477        187
116  206        392
117  687        NaN
118  568        362
119  399         71
120  255        591
121  262        221
122  633        573
123  616        372
124  320        479
125  144         99
126  534        157
127  342        296
128   52        330
129  139        316
130  323        106
131  573        207
132  560        308
133  246        459
134   69        468
135  319        537
136  584        342
137  311        162
138  316        579
139  497        517
140  297        ' '
141   68        135
142  330        118
143  515        341
144  289        329
145  210        570
146  474        320
147  301        335
148  215        NaN
149  606        521
150  590        225

Renomear as colunas de `material_didatico` e aplicar o padrão snake case.

In [16]:
column_mapping = {
    'id': 'id',
    'Quantidade': 'qt_material_didatico'
}

material_didatico.rename(columns=column_mapping,inplace=True)
material_didatico

id qt_material_didatico
0    178                   20
1    634                  121
2    483                  220
3    476                  190
4    132                   67
5    410                  397
6     55                   92
7    490                  578
8    600                  591
9     89                  473
10   387                  456
11   331                  223
12   450                  420
13   558                  573
14   413                   ''
15   698                  366
16   540                  516
17   309                  319
18   268                  308
19   523                  111
20    17                   48
21    35                  525
22    11                  445
23   226                  164
24   638                  293
25   271                  152
26   126                  399
27   180                  185
28   163                  576
29   361                   63
30   280                  491
31    60                'NaN'
32   587                  336
33   653                  159
34   660                  342
35   382                  262
36   217                  440
37   378                  365
38   429                  NaN
39    25                  183
40   367                  297
41   373                   44
42   624                  109
43   157                  350
44    20                  255
45   117                  457
46   502                  493
47   351                  489
48   135                  456
49   647                   64
50   440                  530
51   259                  158
52   422                  123
53    41                  468
54   537                  234
55   230                   24
56   683                  218
57   146                  151
58   566                  564
59   388                  504
60   498                   61
61   663                  426
62   121                  167
63   161                  509
64   263                  292
65   118                  210
66   140                  153
67   260                '397'
68   236                   41
69    90                   67
70   274                  561
71   676                  180
72   390                  257
73   608                  260
74   589                  317
75   256                  345
76    91                'NaN'
77   285                   32
78   248                  513
79   441                  338
80   623                  416
81   603                  352
82   673                  393
83   182                  425
84   520                  136
85   412                '395'
86    31                  115
87   514                  534
88    23                  342
89   415                  152
90    16                  430
91   570                  466
92   528                   51
93   414                   26
94   690                   94
95   202                  121
96   550                  128
97   211                  520
98   526                  524
99   492                  NaN
100  466                  279
101  636                  161
102  527                  179
103  586                  194
104   80                  599
105  398                  315
106  175                  435
107  308                  400
108  512                  294
109  576                  148
110  495                  128
111  379                  109
112  641                  269
113  689                  571
114  171                  235
115  477                  187
116  206                  392
117  687                  NaN
118  568                  362
119  399                   71
120  255                  591
121  262                  221
122  633                  573
123  616                  372
124  320                  479
125  144                   99
126  534                  157
127  342                  296
128   52                  330
129  139                  316
130  323                  106
131  573                  207
132  560        

Mudar o tipo de dados da coluna `id`para string e preencher com zeros (à esquerda) até len = 3.

In [17]:
material_didatico['id'] = material_didatico['id'].astype(str).str.zfill(3)
material_didatico

id qt_material_didatico
0    178                   20
1    634                  121
2    483                  220
3    476                  190
4    132                   67
5    410                  397
6    055                   92
7    490                  578
8    600                  591
9    089                  473
10   387                  456
11   331                  223
12   450                  420
13   558                  573
14   413                   ''
15   698                  366
16   540                  516
17   309                  319
18   268                  308
19   523                  111
20   017                   48
21   035                  525
22   011                  445
23   226                  164
24   638                  293
25   271                  152
26   126                  399
27   180                  185
28   163                  576
29   361                   63
30   280                  491
31   060                'NaN'
32   587                  336
33   653                  159
34   660                  342
35   382                  262
36   217                  440
37   378                  365
38   429                  NaN
39   025                  183
40   367                  297
41   373                   44
42   624                  109
43   157                  350
44   020                  255
45   117                  457
46   502                  493
47   351                  489
48   135                  456
49   647                   64
50   440                  530
51   259                  158
52   422                  123
53   041                  468
54   537                  234
55   230                   24
56   683                  218
57   146                  151
58   566                  564
59   388                  504
60   498                   61
61   663                  426
62   121                  167
63   161                  509
64   263                  292
65   118                  210
66   140                  153
67   260                '397'
68   236                   41
69   090                   67
70   274                  561
71   676                  180
72   390                  257
73   608                  260
74   589                  317
75   256                  345
76   091                'NaN'
77   285                   32
78   248                  513
79   441                  338
80   623                  416
81   603                  352
82   673                  393
83   182                  425
84   520                  136
85   412                '395'
86   031                  115
87   514                  534
88   023                  342
89   415                  152
90   016                  430
91   570                  466
92   528                   51
93   414                   26
94   690                   94
95   202                  121
96   550                  128
97   211                  520
98   526                  524
99   492                  NaN
100  466                  279
101  636                  161
102  527                  179
103  586                  194
104  080                  599
105  398                  315
106  175                  435
107  308                  400
108  512                  294
109  576                  148
110  495                  128
111  379                  109
112  641                  269
113  689                  571
114  171                  235
115  477                  187
116  206                  392
117  687                  NaN
118  568                  362
119  399                   71
120  255                  591
121  262                  221
122  633                  573
123  616                  372
124  320                  479
125  144                   99
126  534                  157
127  342                  296
128  052                  330
129  139                  316
130  323                  106
131  573                  207
132  560        

In [23]:
material_didatico1 = material_didatico

material_didatico1['qt_material_didatico'] = material_didatico1['qt_material_didatico'].fillna(0)
material_didatico1['qt_material_didatico'] = material_didatico1['qt_material_didatico'].str.replace(r'', '0',regex=True)
# material_didatico1['qt_material_didatico'] = material_didatico1['qt_material_didatico'].astype(int)

material_didatico1

id qt_material_didatico
0    178                02000
1    634              0102010
2    483              0202000
3    476              0109000
4    132                06070
5    410              0309070
6    055                09020
7    490              0507080
8    600              0509010
9    089              0407030
10   387              0405060
11   331              0202030
12   450              0402000
13   558              0507030
14   413                0'0'0
15   698              0306060
16   540              0501060
17   309              0301090
18   268              0300080
19   523              0101010
20   017                04080
21   035              0502050
22   011              0404050
23   226              0106040
24   638              0209030
25   271              0105020
26   126              0309090
27   180              0108050
28   163              0507060
29   361                06030
30   280              0409010
31   060          0'0N0a0N0'0
32   587              0303060
33   653              0105090
34   660              0304020
35   382              0206020
36   217              0404000
37   378              0306050
38   429                  NaN
39   025              0108030
40   367              0209070
41   373                04040
42   624              0100090
43   157              0305000
44   020              0205050
45   117              0405070
46   502              0409030
47   351              0408090
48   135              0405060
49   647                06040
50   440              0503000
51   259              0105080
52   422              0102030
53   041              0406080
54   537              0203040
55   230                02040
56   683              0201080
57   146              0105010
58   566              0506040
59   388              0500040
60   498                06010
61   663              0402060
62   121              0106070
63   161              0500090
64   263              0209020
65   118              0201000
66   140              0105030
67   260          0'0309070'0
68   236                04010
69   090                06070
70   274              0506010
71   676              0108000
72   390              0205070
73   608              0206000
74   589              0301070
75   256              0304050
76   091          0'0N0a0N0'0
77   285                03020
78   248              0501030
79   441              0303080
80   623              0401060
81   603              0305020
82   673              0309030
83   182              0402050
84   520              0103060
85   412          0'0309050'0
86   031              0101050
87   514              0503040
88   023              0304020
89   415              0105020
90   016              0403000
91   570              0406060
92   528                05010
93   414                02060
94   690                09040
95   202              0102010
96   550              0102080
97   211              0502000
98   526              0502040
99   492                  NaN
100  466              0207090
101  636              0106010
102  527              0107090
103  586              0109040
104  080              0509090
105  398              0301050
106  175              0403050
107  308              0400000
108  512              0209040
109  576              0104080
110  495              0102080
111  379              0100090
112  641              0206090
113  689              0507010
114  171              0203050
115  477              0108070
116  206              0309020
117  687                  NaN
118  568              0306020
119  399                07010
120  255              0509010
121  262              0202010
122  633              0507030
123  616              0307020
124  320              0407090
125  144                09090
126  534              0105070
127  342              0209060
128  052              0303000
129  139              0301060
130  323              0100060
131  573              0200070
132  560        

In [19]:
duplicates_mask = material_didatico.duplicated(subset='id',keep=False)
duplicates_df = material_didatico[duplicates_mask]
duplicates_df.sort_values('id')

Empty DataFrame
Columns: [id, qt_material_didatico]
Index: []

In [20]:
subprefeituras = pd.read_csv('data/subprefeituras.csv')

column_mapping = {
    'nome': 'bairro',
    'subprefeitura': 'subprefeitura'
}

subprefeituras.rename(columns=column_mapping,inplace=True)
subprefeituras['bairro'] = subprefeituras['bairro'].str.title()
subprefeituras

bairro    subprefeitura
0                Santo Cristo           Centro
1                      Gamboa           Centro
2                       Saúde           Centro
3                        Caju           Centro
4                      Centro           Centro
5                        Lapa           Centro
6                Rio Comprido           Centro
7                 Cidade Nova           Centro
8                     Catumbi           Centro
9                     Estácio           Centro
10                  Mangueira           Centro
11                    Benfica           Centro
12              Vasco Da Gama           Centro
13              São Cristóvão           Centro
14                    Paquetá            Ilhas
15               Santa Teresa           Centro
16                     Glória           Centro
17                Cosme Velho         Zona Sul
18                   Botafogo         Zona Sul
19                   Flamengo         Zona Sul
20                    Humaitá         Zona Sul
21                       Urca         Zona Sul
22                Laranjeiras         Zona Sul
23                     Catete         Zona Sul
24                       Leme         Zona Sul
25                 Copacabana         Zona Sul
26                    Vidigal         Zona Sul
27                      Lagoa         Zona Sul
28            Jardim Botânico         Zona Sul
29                    Ipanema         Zona Sul
30                São Conrado         Zona Sul
31                      Gávea         Zona Sul
32                     Leblon         Zona Sul
33                    Rocinha         Zona Sul
34          Praça Da Bandeira    Grande Tijuca
35                     Tijuca    Grande Tijuca
36          Alto Da Boa Vista    Grande Tijuca
37                   Maracanã    Grande Tijuca
38                    Andaraí    Grande Tijuca
39                     Grajaú    Grande Tijuca
40                Vila Isabel    Grande Tijuca
41                 Manguinhos       Zona Norte
42                     Olaria       Zona Norte
43                 Bonsucesso       Zona Norte
44                      Ramos       Zona Norte
45                       Maré       Zona Norte
46                   Abolição       Zona Norte
47        Lins De Vasconcelos       Zona Norte
48                    Piedade       Zona Norte
49                     Jacaré       Zona Norte
50                   Cachambi       Zona Norte
51                 Água Santa       Zona Norte
52            Todos Os Santos       Zona Norte
53                    Pilares       Zona Norte
54                  Riachuelo       Zona Norte
55                      Méier       Zona Norte
56                    Sampaio       Zona Norte
57       São Francisco Xavier       Zona Norte
58                      Rocha       Zona Norte
59                  Encantado       Zona Norte
60          Engenho De Dentro       Zona Norte
61               Engenho Novo       Zona Norte
62                Jacarezinho       Zona Norte
63        Vicente De Carvalho       Zona Norte
64                    Colégio       Zona Norte
65                Vila Kosmos       Zona Norte
66               Vista Alegre       Zona Norte
67              Vila Da Penha       Zona Norte
68                      Irajá       Zona Norte
69                  Cascadura       Zona Norte
70                 Cavalcanti       Zona Norte
71                    Turiaçú       Zona Norte
72              Bento Ribeiro       Zona Norte
73               Osvaldo Cruz       Zona Norte
74              Rocha Miranda       Zona Norte
75            Engenheiro Leal       Zona Norte
76                   Campinho       Zona Norte
77                   Vaz Lobo       Zona Norte
78            Marechal Hermes       Zona Norte
79                  Madureira       Zona Norte
80          Quintino Bocaiúva       Zona Norte
81             Honório Gurgel       Zona Norte
82          Engenho Da Rainha       Zona Norte
83               Del Castilho       Zona Norte
84             Maria Da Graç

In [21]:
duplicates_mask = subprefeituras.duplicated(subset='bairro',keep=False)
duplicates_df = subprefeituras[duplicates_mask]
duplicates_df.sort_values('bairro')

Empty DataFrame
Columns: [bairro, subprefeitura]
Index: []

In [22]:
df_temp = pd.merge(escolas_unique,material_didatico,on='id', how='left')

df_temp

NameError: name 'escolas_unique' is not defined

In [ ]:
df_final = pd.merge(df_temp,subprefeituras,on='bairro',how='left')

df_final

In [ ]:
df_final.describe()

In [ ]:
df_final.info()

In [ ]:
df_final[df_final['subprefeitura'].isnull()]

In [ ]:
qt_mask = pd.to_numeric(df_final['qt_material_didatico'], errors='coerce').isna()
not_numeric = df_final[qt_mask]
not_numeric

In [ ]:
subprefeituras_lst = list(subprefeituras['bairro'].unique())
subprefeituras_lst.sort()
subprefeituras_lst